keadm/cmd/keadm/app/cmd/ctl/describe/device.go

```diff
@@ -0,0 +1,195 @@
+	cmd.Flags().BoolVar(&options.ShowEvents, common.FlagNameShowEvents, false, "If present, list the requested object(s) across all namespaces")
+	cmd.Flags().Int64Var(&options.ChunkSize, common.FlagNameChunkSize, 500, "If non-zero, split output into chunks of this many bytes")
+}
+
+func (o *DeviceDescribeOptions) describeDevice(args []string) error {
+	config, err := util.ParseEdgecoreConfig(common.EdgecoreConfigPath)
+	if err != nil {
+		return fmt.Errorf("get edge config failed with err:%v", err)
+	}
+	nodeName := config.Modules.Edged.HostnameOverride
+
+	ctx := context.Background()
+
+	var deviceListFilter *v1beta1.DeviceList
+
+	if len(args) > 0 {
+		deviceListFilter = &v1beta1.DeviceList{
+			Items: make([]v1beta1.Device, 0, len(args)),
+		}
+
+		var deviceRequest *client.DeviceRequest
+		for _, deviceName := range args {
+			deviceRequest = &client.DeviceRequest{
+				Namespace:  o.Namespace,
+				DeviceName: deviceName,
+			}
+			device, err := deviceRequest.GetDevice(ctx)
+			if err != nil {
+				klog.Error(err.Error())
+				continue
+			}
+			if device.Spec.NodeName == nodeName {
+				deviceListFilter.Items = append(deviceListFilter.Items, *device)
+			} else {
+				klog.Infof("can't query device: \"%s\" for node: \"%s\"", device.Name, device.Spec.NodeName)
+			}
+		}
+	} else {
+		deviceRequest := &client.DeviceRequest{
+			Namespace:     o.Namespace,
+			LabelSelector: o.LabelSelector,
+			AllNamespaces: o.AllNamespaces,
+		}
+
+		deviceList, err := deviceRequest.GetDevices(ctx)
+		if err != nil {
+			return err
+		}
+
+		deviceListFilter = &v1beta1.DeviceList{
+			Items: make([]v1beta1.Device, 0, len(deviceList.Items)),
+		}
+
+		for _, device := range deviceList.Items {
+			if device.Spec.NodeName == nodeName {
+				deviceListFilter.Items = append(deviceListFilter.Items, device)
+			}
+		}
+	}
+
+	if len(deviceListFilter.Items) == 0 {
+		if len(args) > 0 {
+			return nil
+		}
+
+		if o.AllNamespaces {
+			klog.Info("No resources found in all namespaces.")
+		} else {
+			klog.Infof("No resources found in %s namespaces.", o.Namespace)
+		}
+
+		return nil
+	}
+
+	NamespaceToDeviceName := make(map[string][]string)
+
+	for _, device := range deviceListFilter.Items {
+		if _, ok := NamespaceToDeviceName[device.Namespace]; !ok {
+			NamespaceToDeviceName[device.Namespace] = make([]string, 0)
+		}
+		NamespaceToDeviceName[device.Namespace] = append(NamespaceToDeviceName[device.Namespace], device.Name)
+	}
+
+	m := &meta.RESTMapping{
+		Resource:         v1beta1.SchemeGroupVersion.WithResource("devices"),
+		GroupVersionKind: v1beta1.SchemeGroupVersion.WithKind("Device"),
+		Scope:            meta.RESTScopeNamespace,
+	}
+	c, err := client.GetKubeConfig()
+	if err != nil {
+		return err
+	}
+	d, ok := describe.GenericDescriberFor(m, c)
+	if !ok {
+		return fmt.Errorf("unable to find describer for %v", m)
+	}
+
+	first := true
```

Comment:
What's the actual effect of `first`?

Reply:
`describe` supports printing information about multiple resources. If the `first` is false, blank lines is printed to separate resources.